packages/neovim/src/api/Neovim.ts

```diff
@@ -232,7 +232,7 @@ export class Neovim extends BaseApi {
   /**
    * Sets the current tabpage
    */
-  set tabpage(tabpage: AsyncTabpage) {
+  set tabpage(tabpage: AsyncTabpage | Tabpage) {
     this.request(`${this.prefix}set_current_tabpage`, [tabpage]);
```

Comment:
I wonder if this and the buffer() case were a hidden bug. Do they need the same treatment as in window()? I don't see anywhere in `this.request` codepath that would implicitly handle a Promise (AsyncBuffer/AsyncTabpage)

```ts
    if (tabpage instanceof Tabpage) this.request(`${this.prefix}set_current_tabpage`, [tabpage]);
    else tabpage.then(tabpage => this.request(`${this.prefix}set_current_tabpage`, [tabpage]));
```